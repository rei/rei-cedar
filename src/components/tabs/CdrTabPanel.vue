<script>
import {
  computed, defineComponent, inject, useCssModule,
} from 'vue';
import kebabCase from 'lodash-es/kebabCase';

export default defineComponent({
  name: 'CdrTabPanel',
  props: {
    name: String,
  },
  setup(props) {
    const selectedTabName = inject('selectedTabName');

    const isActive = computed(() => props.name === selectedTabName?.value);
    const panelId = computed(() => `${kebabCase(props.name)}-panel`);
    const style = useCssModule();
    return {
      selectedTabName,
      isActive,
      panelId,
      style,
    };
  },
});
</script>

<template>
  <section
    v-show="isActive"
    tabindex="0"
    :aria-hidden="!isActive"
    role="tabpanel"
    :class="style['cdr-tab-panel']"
    :id="panelId"
    :aria-labelledby="name"
  >
    <slot />
  </section>
</template>

<style lang="scss" module src="./styles/CdrTabPanel.module.scss">
</style>
