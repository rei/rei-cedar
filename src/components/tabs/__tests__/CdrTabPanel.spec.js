import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrTabPanel from 'componentdir/tabs/CdrTabPanel';
// import CdrTabs from 'componentdir/tabs/CdrTabs';
// import Vue from 'vue';


describe('CdrTabPanel', () => {
  it('renders tab', () => {
    const wrapper = mount(CdrTabPanel, {
      propsData: {
        name: 'test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('is not active by default', () => {
    const wrapper = mount(CdrTabPanel, {
      propsData: {
        name: 'test',
      },
    });
    expect(wrapper.vm.isActive).toBe(false);
  });

  it('is active when the selectedTabName is the same as the tabPanel name', async () => {
    const wrapper = mount(CdrTabPanel, {
      propsData: {
        name: 'test',
      },
      global: {
        provide: {
          selectedTabName: {value: 'test'}
        }
      }
    });
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.isActive).toBe(true);
  });

  it('is NOT active when the selectedTabName is not the same as the tabPanel name', async () => {
    const wrapper = mount(CdrTabPanel, {
      propsData: {
        name: 'test',
      },
      global: {
        provide: {
          selectedTabName: {value: 'nameofsomedifferentab'}
        }
      }
    });
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.isActive).toBe(false);
  });

});
