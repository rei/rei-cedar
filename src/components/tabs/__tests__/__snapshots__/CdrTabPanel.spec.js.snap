// Vitest Snapshot v1

exports[`CdrTabPanel > default config > renders as expected 1`] = `
<section
  aria-hidden="true"
  aria-labelledby="test"
  class="cdr-tab-panel"
  id="test-panel"
  role="tabpanel"
  style="display: none;"
  tabindex="0"
>
  
  
</section>
`;

exports[`CdrTabPanel > when the selectedTabName is NOT the same as the tabPanel name > renders as expected 1`] = `
<section
  aria-hidden="true"
  aria-labelledby="test"
  class="cdr-tab-panel"
  id="test-panel"
  role="tabpanel"
  style="display: none;"
  tabindex="0"
>
  
  
</section>
`;

exports[`CdrTabPanel > when the selectedTabName is the same as the tabPanel name > renders as expected 1`] = `
<section
  aria-hidden="false"
  aria-labelledby="test"
  class="cdr-tab-panel"
  id="test-panel"
  role="tabpanel"
  tabindex="0"
>
  
  
</section>
`;
